shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1392,6 +1428,38 @@ - (void)updateEditingState {
   }
 }
 
+- (void)updateEditingStateWithDelta:(FlutterTextEditingDelta*)delta {
+  NSUInteger selectionBase = ((FlutterTextPosition*)_selectedTextRange.start).index;
+  NSUInteger selectionExtent = ((FlutterTextPosition*)_selectedTextRange.end).index;
+
+  // Empty compositing range is represented by the framework's TextRange.empty.
+  NSInteger composingBase = -1;
+  NSInteger composingExtent = -1;
+  if (self.markedTextRange != nil) {
+    composingBase = ((FlutterTextPosition*)self.markedTextRange.start).index;
+    composingExtent = ((FlutterTextPosition*)self.markedTextRange.end).index;
+  }
+
+  NSDictionary* deltaToFramework = @{
+    @"oldText" : [NSString stringWithString:delta.oldText],
```

Comment:
these are already copies I think? You should be able to directly use `delta.oldText`?